src/editor/utils/index.js

```diff
@@ -129,9 +130,12 @@ export const deepCopy = object => {
   return obj
 }
 
-export const loadImage = url => {
-  const image = new Image()
+export const loadImage = async (url, detectContentType) => {
+  if (detectContentType && await !checkImageContentType(url)) {
```

Comment:
Do not use `await` statements in if condition, you can use a variable to cache this `await` statement result.

```javascript
if (detectContentType) {
    const isImage = await checkImageContentType(url)
    if (!isImage) return Promise.reject('not an image')
}
return new Promise((resolve, reject) => { // other codes })
// other codes
```